[ot] Define (unused) OTP scrambling properties for EG

jwnrt opened the pull request:
Earl Grey does not support OTP scrambling and does not have these parameters declared. QEMU complains when they are provided by the INI file generated by `cfggen.py.

This PR declares the properties but does not use them to silence this warning. It also removes the default values set in `ot_{darjeeling,earlgrey}` to avoid silent misconfigurations if these change.

I did not notice that this was an issue before because I've been erroneously using `-T earlgrey` and not `EarlGrey` so the table was called `ot-otp-e` and wasn't being used.

rivos-eblot commented:
> I did not notice that this was an issue before because I've been erroneously using -T earlgrey and not EarlGrey so the table was called ot-otp-e and wasn't being used.

Have you tried with the latest `cfggen.py` version? There was a leftover `opentitan[0]` in the initial change for #117 but it has been addressed before the merge. If not, there is another issue :-(

rivos-eblot commented:
> Temporarily disable these constants for Earl Grey.

Would it be better to define and not use yet those properties in ot_otp_eg rather than adding a special case in the script (I may have miss something though)?

jwnrt commented:
> Have you tried with the latest cfggen.py version?

Yes, here's the generated file from the latest `ot-earlgrey-9.1.0` and OpenTitan master:

```console
$ scripts/opentitan/cfggen.py ~/projects/opentitan -T earlgrey -o cfg.ini
```
```ini
[ot_device "ot-otp-eg"]
  digest_const = "e95f517cb98955b4d5a89aa9109294a"
  digest_iv = "bead91d5fa4e0915"
  sram_const = "4a22d4b78fe0266fbee3958332f2939b"
  sram_iv = "f98c48b1f9377284"

[ot_device "ot-lc_ctrl"]
  lc_state_first = "ee75b407d2314d2ef84185ac8c990f536071632c086d4c924070be92d2948d6228b2711e9b2d8c4d"
  lc_state_last = "ee75fe0ffe7b6f3ffc5f9ffd9ff96fdb7f736f6c9e6fdcd35277fef2d3bdcd6ffbb2f59fdf3fbedd"
  lc_trscnt_first = "dfb6c45a241f85ce9f42229e8627462fdb02c6701242f14b41891180045c09c26c52744267c04aa055921b9461bb07da"
  lc_trscnt_last = "dfb6f4fabf1fefcebf5ba2ffc677c6afdbabcefeb672f36b4fbdb3988dfe1be67e7e77ca77c76af7ddde3b9e7fbfe7de"
  ownership_first = "3c61398d626885931da660ed2ab18a0f"
  ownership_last = "bc757dbdeaf9b7d35de7e9efabfbbecf"
  raw_unlock_token = "ea2b3f32cbe77554e43c8ea7ebf197c2"
  socdbg_first = "440af80d"
  socdbg_last = "6c9ef9cf"
```

I also noticed that:

* the `ot-rom_ctrl` table has been removed, is that intentional?
* the `ot-otp-eg.scrmbl_key` field has been removed. This might be due to upstream OpenTitan changes moving this to another file?

jwnrt commented:
> > Temporarily disable these constants for Earl Grey.
> 
> Would it be better to define and not use yet those properties in ot_otp_eg rather than adding a special case in the script (I may have miss something though)?

Yes, this is probably a better solution, I just went with a quicker hack. I can do this properly if you'd prefer.

rivos-eblot commented:
> Yes

No sure to what it applies :) Is it ok now?

> * the `ot-rom_ctrl` table has been removed, is that intentional?
> * the `ot-otp-eg.scrmbl_key` field has been removed. This might be due to upstream OpenTitan changes moving this to another file?

No, not intentional. I'll have a look at both these issues.

rivos-eblot commented:
> Yes, this is probably a better solution, I just went with a quicker hack. I can do this properly if you'd prefer.

Since we want to deliver it to the master branch, I think it would be better yes, thanks.
I think it would be also safer to remove the default values in the `ot_earlgrey.c` as well, so that if no config file is provided, OTP fails rather than massaging data with the wrong values, which could cause some debug headaches :). I initially kept them so the command line was simpler, but there are now too many configuration parameters for OTP. This was a bad idea :)

jwnrt commented:
> No sure to what it applies :) Is it ok now?

I've fixed my original comment

jwnrt committed:
[ot] hw/opentitan: otp_ctrl: declare scrambling properties

These are currently unused but we should declare them so that the table
of constants generated by `cfggen.py` continues working.

Signed-off-by: James Wainwright <james.wainwright@lowrisc.org>

jwnrt committed:
[ot] hw/riscv: dj+eg: remove default OTP scrambling values

Providing defaults here could be error prone. Instead, require that
these properties are set by the `cfggen.py` INI file.

There is already a trace in `ot_otp_dj_configure_scrmbl_key` and
`ot_otp_dj_configure_digest` when these properties are not set.

Signed-off-by: James Wainwright <james.wainwright@lowrisc.org>

jwnrt commented:
I've updated the PR, the properties are now declared but not used in EG and I've removed the default values for DJ. I haven't tested DJ, but I can see warnings in `otp_ctrl_dj.c` already so I think this is handled.

rivos-eblot commented on hw/opentitan/ot_otp_eg.c:
```diff
@@ -490,6 +490,12 @@ struct OtOTPEgState {
     OtOtpBeIf *otp_backend;
     OtEDNState *edn;
     uint8_t edn_ep;
+    // OTP scrambling property strings: currently unused, see issue #121.
```
QEMU checkpatch.pl tool (which we barely use :-)) would complain about using C++ comment style whereas only C comment style should be used:
```h
/* single line */

/*
 * multi
 * line
 */
```

rivos-eblot commented on hw/opentitan/ot_otp_eg.c:
```diff
@@ -1214,6 +1220,12 @@ static Property ot_otp_eg_properties[] = {
                      OtOtpBeIf *),
     DEFINE_PROP_LINK("edn", OtOTPEgState, edn, TYPE_OT_EDN, OtEDNState *),
     DEFINE_PROP_UINT8("edn-ep", OtOTPEgState, edn_ep, UINT8_MAX),
+    // OTP scrambling property strings: currently unused, see issue #121.
```
I'm not sure if there is a policy about this, but would it not be safer to use the full GitHub link so that the destination URL is known when we browse history?

rivos-eblot reviewed (approved):
LGTM

AlexJones0 reviewed (approved):
LGTM, thanks @jwnrt.

rivos-eblot commented:
nit pick: maybe update the PR title, as there is no more changes in `cfggen.py`

rivos-eblot commented:
> No, not intentional. I'll have a look at both these issues.

Both are due to a regression I've unfortunately added with #117. Working in on a patch

rivos-eblot commented:
> Both are due to a regression I've unfortunately added with #117. Working in on a patch

#124
